### PR TITLE
Update VariationalAutoencoder.py

### LINKED ISSUE #2243: Cost calculation in Variational Autoencoder model is wrong
### System information
- **What is the top-level directory of the model you are using**: autoencoder
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: no
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 14.04
- **TensorFlow installed from (source or binary)**: not relevant
- **TensorFlow version (use command below)**: not relevant
- **Bazel version (if compiling from source)**:not relevant
- **CUDA/cuDNN version**: not relevant
- **GPU model and memory**: not relevant
- **Exact command to reproduce**: not relevant

### Describe the problem
The [reconstruction error](https://github.com/tensorflow/models/blob/master/autoencoder/autoencoder_models/VariationalAutoencoder.py#L24) in variational autoencoder is calculated without specifying the "axis" argument, resulting in calculating the sum of reconstruction error across the whole batch. However, the [latent error](https://github.com/tensorflow/models/blob/master/autoencoder/autoencoder_models/VariationalAutoencoder.py#L25) is calculated with axis=1, resulting in a 1D vector representing the latent error for each sample in the batch. When we add them up and take the average to get the [cost](https://github.com/tensorflow/models/blob/master/autoencoder/autoencoder_models/VariationalAutoencoder.py#L26), the 0-d reconstruction error will get broadcast to each element of the latent error, resulting in a larger cost than it should be. The right way should be either mean + mean, or mean(vector + vector).

### Source code / logs
N/A

### PR DESCRIPTION
**Pull request by accraze**
Added `axis=1` for reconstruction error calculation. This will avoid taking the sum of cost so that the training objective will no longer be batch-size dependent.

Fixes #2243

### CODE REVIEW

**Comment by karmel** [comment]
CC @snurkabill -- can you take a look?

**Comment by snurkabill** [comment]
hi. currently I can't do anything. Will be able to work on september

**Comment by jaeyounkim** [comment]
@snurkabill Can you check this PR?

**Comment by snurkabill** [comment]
done.
